ghproxy/README.md

```diff
@@ -3,4 +3,18 @@
 ghProxy is a reverse proxy HTTP cache optimized for use with the GitHub API (https://api.github.com).
 It is essentially just a reverse proxy wrapper around [ghCache](/ghproxy/ghcache) with some additional prometheus instrumentation logic to monitor disk usage and push metrics to a prometheus push gateway.
 
-ghProxy is designed to reduce API token usage by allowing many components to share a single ghCache. Note that components must use the same API token to benefit from the cache and avoid clobbering existing cache entries for other tokens.
\ No newline at end of file
+ghProxy is designed to reduce API token usage by allowing many components to share a single ghCache. Note that components must use the same API token to benefit from the cache and avoid clobbering existing cache entries for other tokens.
+
+## How to use
+
+To use ghProxy for your prow instance,
+
+- Create a `StorageClass` using [`gce-ssd-retain_storageclass.yaml`](/prow/cluster/gce-ssd-retain_storageclass.yaml)
```

Comment:
What is the benefit of the storageclass here?

Reply:
It's used in the [PVC mentioned below](https://github.com/kubernetes/test-infra/blob/master/prow/cluster/ghproxy_deployment.yaml#L35)

Reply:
Okay sorry, let me rephrase my question: What is the advantage of using a Storageclass with `reclaimPolicy: Retain` here?

Reply:
If you redeploy `ghproxy` and you end up reclaiming the data by a new deployment, we want the data to not have been deleted.

Reply:
Doesn't that require manual intervention to produce that effect? If the PVC was deleted, at least the `PersistentVolume.Spec.ClaimRef.UID` field [needs to be cleared]((https://github.com/kubernetes/kubernetes/issues/48609#issuecomment-314066616)) to make it's available again to be bound to the same claim.

AFAIK, `reclaimPolicy: retain` only means that the PV is not deleted when the PVC is deleted. If the PVC is recreated then a new PV is provisioned (and a new persistent desk on GCP). If the PVC is not deleted and only the deployment is recreated, then the PVC should be bound to the same PV regardless of the `reclaimPolicy` set.

Reply:
Interesting. My understanding of PV binding is definitely shakey. @cjwagner thoughts?

Reply:
Yeah that makes a lot of sense. I guess we could remove that and just keep the storage class for the SSD type?